dev-workflow-ui/src/ch/ivyteam/workflowui/DefaultFramePageHandler.java

```diff
@@ -0,0 +1,31 @@
+package ch.ivyteam.workflowui;
+
+import java.io.IOException;
+
+import ch.ivyteam.ivy.application.IApplication;
+import ch.ivyteam.ivy.dialog.execution.api.DialogInstance;
+import ch.ivyteam.ivy.request.IHttpResponse;
+import ch.ivyteam.workflowui.starts.CustomFieldsHelper;
+import ch.ivyteam.workflowui.tasks.TaskModel;
+
+public class DefaultFramePageHandler {
+
+  public static void handleRedirect(String relativeUrl) throws IOException {
+    var task = DialogInstance.of(relativeUrl).task();
+    var url = relativeUrl;
+    if (shouldOpenInFrame(new TaskModel(task))) {
+      url = IApplication.current().getContextPath() + "/faces/view/dev-workflow-ui/frame.xhtml?taskUrl=" + relativeUrl;
+    }
+    IHttpResponse.current().sendRedirect(url);
+  }
+
+  private static boolean shouldOpenInFrame(TaskModel task) {
+    var taskEmbedField = new CustomFieldsHelper(task).getEmbedInFrame();
+    var caseEmbedField = task.getBusinessCase().customFields().stringField(CustomFieldsHelper.EMBED_IN_FRAME);
+    if (taskEmbedField == null) {
+      return "true".equals(caseEmbedField.getOrNull());
+    }
+    return "true".equals(taskEmbedField.getValue());
```

Comment:
as said yesterday:
1. please make sure that this logic does not fail when no task can be resolved
2. it is not about the "true" value which is interesting, when the DefaultFramePage start is called the default behaviour is that it should be embed in frame. It is about escape this default behaviour and open the task full screen. So you should check here if it should not be opened in Frame.